### PR TITLE
fix: 修复 carousel dotposition 问题

### PR DESCRIPTION
**Pull request by xrkffgg**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
close https://github.com/ant-design/ant-design/issues/20638

### 💡 Background and solution
自己改动了一下，不知道是不是符合 4.0 的设计思路

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Fix the display error of carousel component with dotposition of left and right          |
| 🇨🇳 Chinese | 修复carousel 组件 dotposition 为 left 、right 的显示错误          |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/carousel/index.tsx [hunk 1]
```diff
@@ -107,10 +107,14 @@ export default class Carousel extends React.Component<CarouselProps, {}> {
...
     props.dotsClass = `${dotsClass} ${dotsClass}-${dotPosition || 'bottom'}`;
 
-    const className = classNames(prefixCls, {
+    let className = classNames(prefixCls, {
       [`${prefixCls}-rtl`]: direction === 'rtl',
     });
 
+    if (dotPosition === 'left' || dotPosition === 'right') {
+      className = `${className} ${className}-vertical`;
+    }
```
Comment (afc163) [inline components/carousel/index.tsx]:
```
const className = classNames(prefixCls, {
  [`${prefixCls}-rtl`]: direction === 'rtl',
  [`${className}-vertical`]: dotPosition === 'left' || dotPosition === 'right',
});
```

**Comment by afc163** [comment]
需要更新一下 snapshots

File: components/carousel/index.tsx (hunk 1 above)
Comment (xrkffgg) [reply components/carousel/index.tsx]: ok

**Comment by afc163** [comment]
rebase 一下 4.0-prepare

**Comment by xrkffgg** [comment]
以前没有 `rebase` 操作过，不知道现在这样对不对 😂😂
@afc163

File: components/button/index.zh-CN.md [hunk 2]
```diff
@@ -11,6 +11,20 @@ subtitle: 按钮
...
+- 默认按钮：用于没有主次之分的一组行动点。
+- 虚线按钮：常用于添加操作。
+- 链接按钮：用于次要或外链的行动点。
+
+以及四种状态属性与上面配合使用。
+
+- 危险：删除/移动/修改权限等危险操作，一般需要二次确认。
+- 幽灵：用于背景色比较复杂的地方，常用在首页/产品页等展示场景。
+- 禁用：行动点不可用的时候，一般需要文案解释。
+- 加载中：用于异步操作等待反馈的时候，也可以避免多次提交。
```
Comment (afc163) [inline components/button/index.zh-CN.md]: rebase 失败

Comment (xrkffgg) [reply components/button/index.zh-CN.md]: 尴尬，我重新拉取了最新的分支代码，新建了另一个分支改了

Comment (xrkffgg) [reply components/button/index.zh-CN.md]: 我用最新的 4.0 再建个分支 改了

Comment (xrkffgg) [reply components/button/index.zh-CN.md]: https://github.com/ant-design/ant-design/pull/20645

还不行的话，就把这 2 个 PR 都关了，你们改一下吧😂

Comment (xrkffgg) [reply components/button/index.zh-CN.md]: @afc163 大佬，你刚才让我 `rebase`，
是想让我把  两个 `commit` 合并成一个？
还是要 同步 4.0-prepare 更新代码？

Comment (afc163) [reply components/button/index.zh-CN.md]: `git rebse 4.0-prepare`

**Pull request closed without merging**